src-tauri/src/main.rs

```diff
@@ -156,26 +153,42 @@ async fn main() {
                 // and move it to another function or thread
                 let item_handle = app.tray_handle().get_item(&id);
                 match id.as_str() {
-                    "hide" => {
+                    "hide_show" => {
                         let window = app.get_window("main").unwrap();
-                        window.hide().unwrap();
+
                         // you can also `set_selected`, `set_enabled` and `set_native_image` (macOS only).
-                        item_handle.set_title("Show").unwrap();
+                        match window.is_visible().unwrap() {
+                            true => {
+                                window.hide().unwrap();
+                                item_handle.set_title("Show").unwrap()
+                            }
+                            false => {
+                                window.show().unwrap();
+                                item_handle.set_title("Hide").unwrap()
```

Comment:
I think it would be good to use `.expect('Message with some context')` instead of `.unwrap`. As if in the rare scenario that it fails at the unwrap it'll give you some more contextual information rather than a pretty generic error

Reply:
Fixed in: 028eef9